spec/features/cdr/cdr_history/cdr_filter_spec.rb

```diff
@@ -0,0 +1,72 @@
+RSpec.describe 'Cdrs index page filtering', js: true do
+  subject do
+    within_filters { click_submit('Filter') }
+  end
+
+  include_context :login_as_admin
+
+  before do
+    Cdr::Cdr.delete_all
+    visit cdrs_path
+
+    within_filters do
+      select_by_value tagged_filter_value, from: 'Tagged'
+    end
+  end
+
+  let!(:cdrs) { create_list(:cdr, 2) }
+  let!(:tag) { create :routing_tag }
+  let!(:cdr_tagged) { create :cdr, routing_tag_ids: [tag.id] }
+  let(:tagged_filter_value) { nil }
+
+  context 'with filter by tagged' do
+    let(:tagged_filter_value) { true }
+
+    it 'shows filtered records with routing_tags' do
+      subject
+      expect(page).to have_table
+      expect(page).to have_table_row count: 1
+      expect(page).to have_table_cell column: 'Id', text: cdr_tagged.id
+      expect(page).to have_select :Tagged, selected: 'Yes'
+    end
+  end
+
+  context 'with filter by not tagged' do
+    let(:tagged_filter_value) { false }
+
+    it 'shows filtered records without routing_tags' do
+      subject
+      expect(page).to have_table
+      expect(page).to have_table_row count: 2
+      expect(page).to have_table_cell column: 'Id', text: cdrs.first.id
```

Comment:
this test does not check that tagged cdr does not appear on page. 

to test it you could add just

```ruby
expect(page).to_not have_table_cell column: 'Id', text: cdr_tagged.id
```
but I suggest another approach: we can check that only correct records are shown:

```ruby
expect(page).to have_table_row count: cdrs.size
cdrs.each { |cdr| expect(page).to have_table_cell column: 'Id', text: cdr.id }
```

this method is more readable and stable